usr/src/cmd/svc/milestone/fs-joyent

```diff
@@ -70,6 +70,22 @@ function mount_zfs
     echo ${output} 1>&2
 }
 
+function unlock_pool
+{
+    local pool=$1
+
+    # If the key is already loaded, don't bother trying again
+    local keystatus="$(zfs get -Hpo value keystatus $pool)"
+    if [[ "$keystatus" == "available" ]]; then
+        return
+    fi
+
+    kbmadm unlock $pool && return
+
+    echo "Failed to unlock $pool; recovery may be required" | \
```

Comment:
Recovery *may be* or *is* required?  That is, in what cases would it not be required?

It seems this should cause the method script to exit with `$SMF_EXIT_ERR_FATAL`.  Or perhaps only exit fatal if this happens and no system pool is found.

Reply:
If there's a problem with the zpool itself, or if KBMAPI isn't responding, all could be errors that don't require recovery.

Reply:
I thought there was some additional stuff fs-joyent does that could be useful even in a failure, but I'll look again and see if we can just exit at that point.

Reply:
Even if there is more stuff that it could do, it seems as though it has failed to do a key thing that is expected of it. As such `$SMF_EXIT_OK` seems wrong.